src/puppetlabs/trapperkeeper/services/scheduler/scheduler_core.clj

```diff
@@ -19,46 +24,58 @@
         (log/error t (i18n/trs "scheduled job threw error"))
         (throw t)))))
 
+(defn build-executable-job
+  ([f job-name group-name] (build-executable-job f job-name group-name {}))
+  ([f job-name group-name options]
+   (let [job (wrap-with-error-logging f)
+         jdm (org.quartz.JobDataMap.)
+         options (assoc options :job job)
+         _ (.put jdm "jobData" options)
+
+         executable-job (-> (org.quartz.JobBuilder/newJob puppetlabs.trapperkeeper.services.scheduler.job)
+                            (.withIdentity job-name group-name)
+                            (.usingJobData jdm))]
+     (.build executable-job))))
+
 (defn interspaced
-  ; See docs on the service protocol,
-  ; puppetlabs.enterprise.services.protocols.scheduler
-  [n f pool]
-  (let [job (wrap-with-error-logging f)]
-    (-> (at-at/interspaced n job pool)
-        :id  ; return only the ID; do not leak the at-at RecurringJob instance
-        )))
+  [n f ^org.quartz.Scheduler pool group-name]
+  (let [job-name (org.quartz.utils.Key/createUniqueName group-name)
+         job (-> (build-executable-job f job-name group-name {:interval n}))
+         schedule (-> (org.quartz.SimpleScheduleBuilder/simpleSchedule))
+         trigger (-> (org.quartz.TriggerBuilder/newTrigger)
+                     (.withSchedule schedule)
+                     (.startNow)
+                     (.build))]
+     (.scheduleJob pool job trigger)
+     (.getJobKey trigger)))
 
 (defn after
-  ; See docs on the service protocol,
-  ; puppetlabs.enterprise.services.protocols.scheduler
-  [n f pool]
-  (let [job (wrap-with-error-logging f)]
-    (-> (at-at/after n job pool)
-        :id  ; return only the ID; do not leak the at-at RecurringJob instance
-        )))
+  [n f ^org.quartz.Scheduler pool group-name]
+  (let [job-name (org.quartz.utils.Key/createUniqueName group-name)
+        job (build-executable-job f job-name group-name)
+        trigger (-> (org.quartz.TriggerBuilder/newTrigger)
+                    (.startAt (org.quartz.DateBuilder/futureDate n org.quartz.DateBuilder$IntervalUnit/MILLISECOND))
+                    (.build))]
+     (.scheduleJob pool job trigger)
+     (.getJobKey trigger)))
 
 (defn stop-job
   "Gracefully stops the job specified by 'id'."
-  [id pool]
-  (at-at/stop id pool))
+  [^org.quartz.JobKey id ^org.quartz.Scheduler pool]
+  (.deleteJob pool id))
```

Comment:
It ignores jobs that don't exist, and only throws when there is an internal Quartz error.  It actually returns true if the job was deleted, and false otherwise.  I'll update the docstring.